File: fibonacci_reloaded.py
def gen_fib(n):
    a,b = 0,1
    yield a
    for _ in range(n):
        a,b=b,a+b
        yield a
        
def fib(n):
    x = list(gen_fib(n))
    return x[n-1]
        
#OR
def good fibonacci(n):
”””Return pair of Fibonacci numbers, F(n) and F(n-1).””” if n <= 1:
return (n,0) else:
(a, b) = good fibonacci(n−1) return (a+b, a)

File: fibonacci_reloaded.js
function* gen_fib(n) {
  let a = 0
  let b = 1;
  yield a;
  while (n > 1) {
    let tmp = a;
    a = b;
    b += tmp;
    yield a;
    n--;
  }
}

function fib(n) {
  let arr = [];
  let gen = gen_fib(n);
  for (var i of gen) {
    arr.push(i);
  }
  return arr[n-1]
}
